lock-api/src/test/java/com/palantir/lock/client/MultiClientCommitTimestampGetterTest.java

```diff
@@ -0,0 +1,251 @@
+        verify(betaCache, times(2)).lastKnownVersion();
+        verify(betaCache, never()).processGetCommitTimestampsUpdate(any(), any());
+    }
+
+    private void setupServiceAndAssertSanityOfResponse(List<BatchElement<NamespacedRequest, Long>> batch) {
+        Map<Namespace, List<GetCommitTimestampsResponse>> expectedResponseMap = new HashMap<>();
+
+        when(timelockService.getCommitTimestamps(any())).thenAnswer(invocation -> {
+            Map<Namespace, GetCommitTimestampsResponse> commitTimestamps =
+                    getCommitTimestampResponse(invocation.getArgument(0));
+            commitTimestamps.forEach((namespace, response) -> {
+                expectedResponseMap
+                        .computeIfAbsent(namespace, _unused -> new ArrayList())
+                        .add(response);
+            });
+            return commitTimestamps;
+        });
+
+        consumer.accept(batch);
+        assertSanityOfResponse(batch, expectedResponseMap);
+    }
+
+    private void assertSanityOfResponse(
+            List<BatchElement<NamespacedRequest, Long>> batch,
+            Map<Namespace, List<GetCommitTimestampsResponse>> expectedResponseMap) {
+        // all requests should be served
+        assertThat(batch.stream().filter(elem -> !elem.result().isDone()).collect(Collectors.toSet()))
+                .isEmpty();
+
+        Map<Namespace, List<Long>> partitionedResponseMap = batch.stream()
+                .collect(groupingBy(
+                        elem -> elem.argument().namespace(),
+                        Collectors.mapping(elem -> Futures.getUnchecked(elem.result()), toList())));
+
+        assertThat(partitionedResponseMap.keySet()).isEqualTo(expectedResponseMap.keySet());
+        assertCorrectnessOfCompletedRequests(expectedResponseMap, partitionedResponseMap);
+    }
+
+    private static void assertCorrectnessOfCompletedRequests(
+            Map<Namespace, List<GetCommitTimestampsResponse>> expectedResponseMap,
+            Map<Namespace, List<Long>> partitionedResponseMap) {
+        KeyedStream.stream(partitionedResponseMap)
+                .forEach((namespace, commitTsList) ->
+                        assertSanityOfServedTimestamps(commitTsList, expectedResponseMap.get(namespace)));
+    }
+
+    private static void assertSanityOfServedTimestamps(
+            List<Long> commitTsList, List<GetCommitTimestampsResponse> commitTimestampsResponses) {
+        long requestedCommitTsCount = commitTimestampsResponses.stream()
+                .mapToLong(resp -> resp.getInclusiveUpper() - resp.getInclusiveLower() + 1)
+                .sum();
+        assertThat(requestedCommitTsCount).isEqualTo(commitTsList.size());
+        assertThat(ImmutableSet.copyOf(commitTsList)).hasSameSizeAs(commitTsList);
+        assertThat(StreamEx.of(commitTsList)
+                        .pairMap((first, second) -> first > second)
+                        .anyMatch(x -> x))
+                .isFalse();
```

Comment:
Would suggest using `as(...)` clauses to outline what each assertion means. They are, in order, if I'm not mistaken

1. we got as many commit timestamps as we asked for
2. commit timestamps we got have no duplicates
3. responses are in increasing order of timestamp (technically this isn't required given this is a single timestamp batch I think, but it's fine)